prereise/gather/demanddata/transportation_electrification/tests/test_immediate_charging_integration.py

```diff
@@ -14,8 +15,118 @@
 )
 
 
+def test_immediate_charging_ldv_2days():
+    first_two_input_days = [1, 2]
+    kwhmi = 0.242
+
+    _, output_load_sum_list, trip_data = immediate_charging(
+        census_region=1,
+        model_year=2017,
+        veh_range=100,
+        power=6.6,
+        location_strategy=2,
+        veh_type="LDV",
+        filepath=os.path.join(
+            const.data_folder_path,
+            "nhts_census_updated_dwell",
+        ),
+    )
+
+    veh_type = "LDV"
+    daily_vmt = get_total_daily_vmt(trip_data, first_two_input_days, veh_type.lower())
+
+    output_load_sum_array = np.array(output_load_sum_list)
+    daily_vmt_array = np.array(daily_vmt)
+    charging_efficiency = 0.9
+
+    assert (
+        abs(
+            (
+                (output_load_sum_array.sum() * charging_efficiency)
+                / (daily_vmt_array.sum() * kwhmi)
+            )
+            - 1
+        )
+        < 0.03
+    )
+
+
+def test_immediate_charging_mdv_1day():
+    veh_type = "MDV"
+    kwhmi = 2.13
+
+    (
+        _,
+        output_load_sum_list,
+        trip_data,
+    ) = immediate_charging_HDV.immediate_hdv_charging(
+        model_year=2017,
+        veh_range=200,
+        power=80,
+        location_strategy=1,
+        veh_type="MDV",
+        filepath=os.path.join(
+            const.data_folder_path,
+            "fdata_v10st.mat",
+        ),
+        trip_strategy=1,
+    )
+
+    daily_vmt = get_total_daily_vmt(trip_data, [1], veh_type.lower())
+
+    output_load_sum_array = np.array(output_load_sum_list)
+    daily_vmt_array = np.array(daily_vmt)
+    charging_efficiency = 0.95
+
+    assert (
+        abs(
+            (
+                (output_load_sum_array.sum() * charging_efficiency)
+                / (daily_vmt_array.sum() * kwhmi)
+            )
+            - 1
+        )
+        < 0.03
+    )
+
+
+def test_immediate_charging_hdv_1day():
```

Comment:
Maybe we can merge this test with the previous one simply by plugging in different `veh_type` and `kwhmi`?